src/main/java/org/testng/internal/Parameters.java

```diff
@@ -338,12 +385,15 @@ private static void checkParameterTypes(String methodName,
       Class<?>[] parameterTypes, String methodAnnotation, String[] parameterNames)
   {
     int totalLength = parameterTypes.length;
+    int tmp = totalLength;
     for (Class parameterType : parameterTypes) {
       if (INJECTED_TYPES.contains(parameterType)) {
         totalLength--;
       }
     }
-
+    if (tmp != totalLength) {
```

Comment:
We could do away with the need for the `tmp` variable if we resorted to `if (totalLength != parameterTypes.length)` ?

Reply:
Yes, I will fix it.